common/key_value_database/in_memory/in_memory.cpp

```diff
@@ -48,59 +48,57 @@ ExpectedBytes InMemoryTransaction::Read(const string &key) {
 	}
 }
 
-ExpectedBool InMemoryTransaction::Write(const string &key, const vector<uint8_t> &value) {
+Error InMemoryTransaction::Write(const string &key, const vector<uint8_t> &value) {
 	assert(!read_only_);
 	db_.map_[key] = value;
-	return ExpectedBool(true);
+	return error::NoError;
 }
 
-ExpectedBool InMemoryTransaction::Remove(const string &key) {
+Error InMemoryTransaction::Remove(const string &key) {
 	assert(!read_only_);
 	db_.map_.erase(key);
-	return ExpectedBool(true);
+	return error::NoError;
 }
 
 ExpectedBytes KeyValueDatabaseInMemory::Read(const string &key) {
 	ExpectedBytes ret {vector<uint8_t>()};
-	auto result = ReadTransaction([&key, &ret](Transaction &txn) -> ExpectedBool {
+	auto err = ReadTransaction([&key, &ret](Transaction &txn) -> Error {
 		ret = txn.Read(key);
-		return ExpectedBool(true);
+		return error::NoError;
 	});
-	return ret;
+	if (err) {
```

Comment:
How tempted on using the `?:` operator were you? :p

Reply:
Haha, you know what? It did not even occur to me! What has happened to me!? :scream: